src/clarity-icons/clarity-icons-element.ts

```diff
@@ -18,30 +20,45 @@ if (typeof Reflect === "object") {
 
 export function ClarityIconElement() {
     "use strict";
-    return (parentConstructor as any).apply(this, arguments);
-}
 
-(ClarityIconElement as any).observedAttributes = ["shape", "size"];
+    const _this = (parentConstructor as any).apply(this, arguments);
+
+    _this.clrIconUniqId = "_clr_icon_" + clrIconId;
+    clrIconId++;
+
+    return _this;
+}
+(ClarityIconElement as any).observedAttributes = ["shape", "size", "title"];
 
 ClarityIconElement.prototype = Object.create(
     HTMLElement.prototype, {constructor: {configurable: true, writable: true, value: ClarityIconElement}});
 
 ClarityIconElement.prototype.constructor = ClarityIconElement;
 
-const generateIcon = function(element: any, shape: string) {
-    shape = shape.split(/\s/)[0];
+const replaceTitleWithCustom = function(currentShapeTemplate: string, currentTitleAttrVal: string,
+                                        clrIconUniqId: string) {
+    const tempContainer = document.createElement("div");
 
-    if (shape !== element._shape) {
-        element._shape = shape;
+    tempContainer.innerHTML = currentShapeTemplate;
+    const svgElement = tempContainer.querySelector("svg");
 
-        // shape exists in set
-        if (ClarityIconsApi.instance.has(shape)) {
-            element.innerHTML = ClarityIconsApi.instance.get(shape);
-        } else {
-            console.error(`'${shape}' is not found in the Clarity Icons set.`);
-            element.innerHTML = ClarityIconsApi.instance.get("error");
-        }
+    if (!svgElement) {
+        return currentShapeTemplate;
     }
+
+    let titleElement = svgElement.querySelector("title");
+
+    if (titleElement) {
+        titleElement.setAttribute("id", clrIconUniqId);
+        titleElement.innerHTML = currentTitleAttrVal;
```

Comment:
I think you want `textContent` here, not `innerHTML`. It's safer and you're not supporting HTML in SVG titles anyway.